ListLibrary/MyList.cs

```diff
@@ -0,0 +1,519 @@
+        {
+            _items = new T[_defaultCapacity];
+        }
+
+        public MyList(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentException("Capacity should be >= 0");
+            }
+
+            if (capacity >= 0 && capacity < 4)
+            {
+                _items = new T[_defaultCapacity];
+            }
+            else
+            {
+                _items = new T[capacity];
+            }
+        }
+
+        public MyList(IEnumerable<T> collection)
+        {
+            _items = new T[_defaultCapacity];
+
+            foreach (T item in collection)
+            {
+                Add(item);
+            }
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            for (int i = 0; i < _size; i++)
+            {
+                yield return _items[i];
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        #endregion
+
+        #region Add methods
+
+        /// <summary>
+        /// Add to end.
+        /// </summary>
+        /// <param name="item">item</param>
+        public void Add(T item)
+        {
+            AddByIndex(_size, item);
+        }
+
+        /// <summary>
+        /// Add one item to the start of list.
+        /// </summary>
+        /// <param name="item">item</param>
+        public void AddFront(T item)
+        {
+            AddByIndex(0, item);
+        }
+
+        /// <summary>
+        /// Add one item  in pointed possition.
+        /// </summary>
+        /// <param name="item"> item </param>
+        /// <param name="pos"> position </param>
+
+        public void AddByIndex(int pos, T item)
+        {
+            if (pos < 0 || pos > _size)
+            {
+                throw new ArgumentOutOfRangeException("Index is out of range");
+            }
+
+            if (_size == _items.Length)
+            {
+                UpdateCapacity(_size + 1);
+            }
+
+            for (int i = _size; i >= 0; i--)
+            {
+                if (i == pos)
+                {
+                    _items[i] = item;
+                    break;
+                }
+                else
+                {
+                    _items[i] = _items[i - 1];
+                }
+            }
+
+            _size++;
+        }
+
+        public void Add(MyList<T> items)
+        {
+            AddByIndex(_size, items);
+        }
+
+        public void AddFront(MyList<T> items)
+        {
+            AddByIndex(0, items);
+        }
+
+        public void AddByIndex(int pos, MyList<T> items)
+        {
+            var newSize = _size + items.Count;
+
+            if (newSize >= _items.Length)
+            {
+                UpdateCapacity(newSize);
+            }
+
+            _size += items.Count;
+
+            for (int i = _size - 1, j = items.Count - 1; j >= 0; i--)
+            {
+                if (i == pos + j)
```

Comment:
I'd prefer it as separate operation after cycle. In this case you will not need `if` inside `for`

Reply:
fixed